packages/dataviews/src/view-grid.js

```diff
@@ -7,20 +7,223 @@ import classnames from 'classnames';
+				<CompositeRow
+					className="dataviews-view-grid__row"
+					role="row"
+					key={ `row-${ index }` }
+				>
+					{ row }
+				</CompositeRow>
+			) ),
+
+		[ chunkedCells ]
+	);
+
+	return useMemo(
+		() => (
+			<>
+				{ rows }
+				{ resizeListener }
+			</>
+		),
+		[ resizeListener, rows ]
+	);
+}
+
+/**
+ * Create CSS selector for element. We avoid IDs and classes as
+ * they can be unstable across renders.
+ *
+ * @param {string}         context Grid item identifier
+ * @param {Element | null} element Node to get selector for
+ * @return {string | undefined} Selector for element
+ */
+function getFocusSelector( context, element ) {
+	if ( ! element || ! element.parentElement ) return undefined;
+	if ( element.id && element.id === context ) {
+		return `#${ context.replaceAll( '/', '\\/' ) }`;
+	}
+
+	const parent = getFocusSelector( context, element.parentElement );
+	const siblings = Array.from( element.parentElement.children );
+	const index = siblings.indexOf( element ) + 1;
+
+	return `${ parent } > ${ element.nodeName }:nth-child(${ index })`;
+}
+
+function Grid( props = {} ) {
+	const gridRef = useRef( null );
+	const store = useCompositeStore( {
+		focusWrap: 'horizontal',
+		rtl: isRTL(),
+	} );
+	const { id: preferredId, children: cells, ...gridProps } = props;
+	const id = useInstanceId( Grid, 'view-grid', preferredId );
+	const rows = useRows( cells, gridRef );
+	const focusedNodeSelectorRef = useRef( null );
+	const eventControllerRef = useRef( null );
+	const [ hasNoPointerEvents, setHasNoPointerEvents ] = useState( false );
+	const [ pointerIsWithinBounds, setPointerIsWithinBounds ] =
+		useState( false );
+	const setFocusedNode = ( context, node ) => {
+		focusedNodeSelectorRef.current = getFocusSelector( context, node );
+	};
+	const [ lastSelectedItem, setLastSelectedItem ] = useState( null );
+	const context = {
+		...store,
+		setFocusedNode,
+		lastSelectedItem,
+		setLastSelectedItem: ( _id, selected = true ) => {
+			setLastSelectedItem( { id: _id, selected } );
+		},
+		hasNoPointerEvents,
+	};
+
+	useEffect( () => {
+		if ( focusedNodeSelectorRef.current ) {
+			const doc = gridRef.current?.ownerDocument;
+			const node = doc.querySelector( focusedNodeSelectorRef.current );
+			node?.focus();
+		}
+	}, [ rows ] );
+
+	useEffect( () => {
+		if ( eventControllerRef.current ) {
+			eventControllerRef.current.abort( 'Pointer moved' );
+		}
+
+		const doc = gridRef.current?.ownerDocument;
+		if ( ! doc ) return;
+
+		if ( pointerIsWithinBounds ) {
+			const listener = ( { key, type } ) => {
+				if ( key === ( isAppleOS() ? 'Meta' : 'Control' ) ) {
+					setHasNoPointerEvents( type === 'keydown' );
+				}
+			};
+			const controller = new AbortController();
+			controller.signal.addEventListener( 'abort', () => {
+				eventControllerRef.current = null;
+			} );
+			const config = { capture: true, signal: controller.signal };
+			doc.body.addEventListener( 'keydown', listener, config );
```

Comment:
Should we add the events in the node we want instead of the body? Why do we need it like this?